src/views/search/orama.tsx

```diff
@@ -0,0 +1,189 @@
+import type { TypedDocument, Orama, Results, Nullable } from "@orama/orama";
+import { OramaClient, SortByClauseUnion } from "@oramacloud/client";
+import { PostInfo } from "types/PostInfo";
+import { CollectionInfo } from "types/CollectionInfo";
+import { PropsWithChildren } from "components/types";
+import { createContext } from "preact";
+import { useContext } from "preact/hooks";
+import { SearchQuery } from "./search";
+
+const postSchema = {
+	slug: "string",
+	tags: "enum[]",
+	authors: "enum[]",
+	title: "string",
+	excerpt: "string",
+	description: "string",
+	searchMeta: "string",
+	publishedTimestamp: "number",
+} as const;
+
+const collectionSchema = {
+	slug: "string",
+	title: "string",
+	excerpt: "string",
+	description: "string",
+	searchMeta: "string",
+	tags: "enum[]",
+	authors: "enum[]",
+	publishedTimestamp: "number",
+} as const;
+
+type PostDocument = TypedDocument<Orama<typeof postSchema>>;
+type CollectionDocument = TypedDocument<Orama<typeof collectionSchema>>;
+
+export interface SearchContext {
+	postClient: OramaClient;
+	collectionClient: OramaClient;
+}
+
+export const SearchClient = createContext<SearchContext>(undefined as never);
+
+interface OramaClientProviderProps extends PropsWithChildren {
+	params?: Partial<ConstructorParameters<typeof OramaClient>[0]>;
+}
+
+export function OramaClientProvider(props: OramaClientProviderProps) {
+	const postClient = new OramaClient({
+		endpoint: "https://cloud.orama.run/v1/indexes/playful-programming-p9lpvl",
+		api_key: "OLeHrFPWLR0alSSZkMiq4tokMZZNEbDL",
```

Comment:
Let's move the `api_key` and `endpoint` into a `constants` file to keep track of it better